android/app/src/main/java/app/covidshield/receiver/worker/ExposureCheckNotificationWorker.kt

```diff
@@ -85,13 +68,32 @@ class ExposureCheckNotificationWorker (private val context: Context, parameters:
 
             val foregroundInfo = ForegroundInfo(1, notification.build())
             setForeground(foregroundInfo)
+        } catch (exception: Exception) {
+            MetricsService.publishDebugMetric(107.0, context, exception.message ?: "Unknown")
+            return Result.failure()
+        }
+
+        val enIsEnabled = exposureNotificationClient.isEnabled.await()
+        val enStatus = exposureNotificationClient.status.await()
+        if (!enIsEnabled || enStatus.contains(ExposureNotificationStatus.INACTIVATED)) {
```

Comment:
Do we want to leave this check at the top? If EN is off, we should bail as early as possible?

Reply:
I think the goal here is to have the `setForeground()` call as soon as possible. Reading the comment in the medium.com article I linked to above.

Reply:
Right, having it above the try below makes sense. I guess my question is, will the notification do anything that the user would notice? If they've disabled EN for whatever reason, we shouldn't be causing foreground work? Which is why I wonder if this specific check should still be at the top.

Reply:
PR looks good but I am just wondering why do we have to run this background task as a foreground service?

Reply:
It's the recommendation from Google to run Exposure Checks as a foreground service.